CodePush.m

```diff
@@ -468,6 +532,59 @@ - (void)savePendingUpdate:(NSString *)packageHash
 }
 
 /*
+ * This method is checks if a new status update exists (new version was installed, 
+ * or an update failed) and return its details (version label, status).
+ */
+RCT_EXPORT_METHOD(getNewStatusReport:(RCTPromiseResolveBlock)resolve
+                            rejecter:(RCTPromiseRejectBlock)reject)
+{
+    if (didRollback) {
+        // Check if there was a rollback that was not yet reported
+        NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+        NSMutableArray *failedUpdates = [preferences objectForKey:FailedUpdatesKey];
+        if (failedUpdates) {
+            NSDictionary* lastFailedPackage = [failedUpdates lastObject];
```

Comment:
The convention is to put * with the variable name not with the type name.  Also applies below.